_source/_posts/2024-01-24-spring-boot-authorization.md

```diff
@@ -0,0 +1,569 @@
+                  */                                                                                                      
+                 exports.onExecutePostLogin = async (event, api) => {                                                     
+                   const namespace = 'https://menu-api.okta.com';                                                         
+                   if (event.authorization) {                                                                             
+                     api.idToken.setCustomClaim('preferred_username', event.user.email);                                  
+                     api.idToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);                         
+                     api.accessToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);                     
+                   }                                                                                                      
+                 };                                                                                                       
+
+
+                 /**                                                                                                      
+                  * Handler that will be invoked when this action is resuming after an external redirect. If your         
+                  * onExecutePostLogin function does not perform a redirect, this function can be safely ignored.         
+                  *                                                                                                       
+                  * @param {Event} event - Details about the user and the context in which they are logging in.           
+                  * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.  
+                  */                                                                                                      
+                 // exports.onContinuePostLogin = async (event, api) => {                                                 
+                 // };                                                                    
+```
+
+You can list the available actions with the following command:
+
+```shell
+auth0 actions list
+```
+The output will show the deployment status of each action:
+
+```text
+=== dev-avup2laz.us.auth0.com actions
+
+  ID                                    NAME       TYPE        STATUS  DEPLOYED  
+  da49ae42-b5e4-496a-8305-4fff437f813b  Add Roles  post-login  built   ✗         
+```
+
+Note the `DEPLOYED` status is `x`. Go ahead and deploy it using the action ID:
+
+```shell
+auth0 actions deploy \<ACTION_ID\>
+```
+
+Once the action is deployed, you must attach it to the login flow. You can do this with Auth0 [Management API for Actions](https://auth0.com/docs/api/management/v2#!/Actions/patch_bindings):
+
+```shell
+auth0 api patch "actions/triggers/post-login/bindings" \
```

Comment:
what does the `\` in `\<ACTION_ID\>` do? are they needed? looks a bit confusing

Reply:
Removed, escaping not needed.